[18.0][MIG] maintenance_equipment_hierarchy: Migration to 18.0

BhaveshHeliconia opened the pull request.

JordiBForgeFlow committed:
add maintenance_equipment_hierarchy

Commit:
Added translation using Weblate (Spanish)

etobella committed:
[11.0][IMP] maintenance_equipment_hierarchy: set track_visibility on parent_id

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (11 of 11 strings)

Translation: maintenance-11.0/maintenance-11.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-11-0/maintenance-11-0-maintenance_equipment_hierarchy/es/

NuriaXifre committed:
[12.0][MIG] maintenance_equipment_hierarchy

[UPD] Update maintenance_equipment_hierarchy.pot

dalonsod committed:
[IMP] maintenance_equipment_hierarchy: black, isort

dalonsod committed:
[MIG] maintenance_equipment_hierarchy: Migration to 13.0

sanrav committed:
Added translation using Weblate (French (France))

sanrav committed:
Translated using Weblate (French (France))

Currently translated at 33.3% (4 of 12 strings)

Translation: maintenance-13.0/maintenance-13.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_equipment_hierarchy/fr_FR/

baimont committed:
[13.0][FIX TRAD] maintneance_equipment_hierarchy - fr translation was not working

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: maintenance-13.0/maintenance-13.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-13-0/maintenance-13-0-maintenance_equipment_hierarchy/it/

mariadforgeflow committed:
[IMP] maintenance_equipment_hierarchy: black, isort, prettier

mariadforgeflow committed:
[MIG] maintenance_equipment_hierarchy: Migration to 14.0

mariadforgeflow committed:
[IMP] maintenance_equipment_hierarchy: black, isort, prettier

mariadforgeflow committed:
[MIG] maintenance_equipment_hierarchy: Migration to 15.0

Commit:
[UPD] Update maintenance_equipment_hierarchy.pot

albariera committed:
[FIX] maintenance_equipment_hierarchy: fix view archived childs and make field translatable

Commit:
[UPD] Update maintenance_equipment_hierarchy.pot

FrancoMaxime committed:
[MIG] maintenance_equipment_hierarchy: Migration to 16.0

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (12 of 12 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_equipment_hierarchy/it/

baimont committed:
[IMP] maintenance_equipment_hierarchy - set unaccent to false on parent_path as proposed by warning logs

baimont committed:
[16.0][FIX] maintenance_equipment_hierarchy - children smartbutton leads wrongly to parent equipment

Commit:
[UPD] Update maintenance_equipment_hierarchy.pot

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_equipment_hierarchy/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_equipment_hierarchy/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (13 of 13 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_equipment_hierarchy/it/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (13 of 13 strings)

Translation: maintenance-16.0/maintenance-16.0-maintenance_equipment_hierarchy
Translate-URL: https://translation.odoo-community.org/projects/maintenance-16-0/maintenance-16-0-maintenance_equipment_hierarchy/es/

peterdovicovic committed:
Added translation using Weblate (Slovak)

dalonsod committed:
[IMP] maintenance_equipment_hierarchy: pre-commit auto fixes

dalonsod committed:
[MIG] maintenance_equipment_hierarchy: Migration to 17.0

dalonsod committed:
[IMP] maintenance_equipment_hierarchy: added Has Children to search view

BhaveshHeliconia committed:
[MIG] maintenance_equipment_hierarchy: Migration to 18.0

rvalyi commented:
/ocabot migration maintenance_equipment_hierarchy

dalonsod commented on maintenance_equipment_hierarchy/tests/test_maintenance_equipment_hierarchy.py:
```diff
@@ -0,0 +1,57 @@
+from odoo.tests import common
+
+
+class TestMaintenanceEquipmentHierarchy(common.TransactionCase):
```
Could you use `BaseCommon` instead?

dalonsod commented on maintenance_equipment_hierarchy/models/maintenance_equipment.py:
```diff
@@ -0,0 +1,67 @@
+                "default_parent_id": self.id,
+                "parent_id_editable": False,
+            },
+            "domain": [("parent_id", "=", self.id)],
```
could you add a `path` for this?

dalonsod reviewed (commented):
See comments

BhaveshHeliconia commented:
@dalonsod : Please review again. Thanks.

dalonsod commented:
Reviewing commit history I've notice that doesn't include stuff from v17.0 version, could you check? I've found at least two commits of my own authorship (https://github.com/OCA/maintenance/commit/9494d7080f40c60643b05879dd56a5d3542d782c and https://github.com/OCA/maintenance/commit/9a921a8576448277d073109fe776207a0c24dd74) that are not included, and it shouldn't be merged in another one without respecting authorship.

In other words, this migration should start from the complete commit history of v17 addon version.

BhaveshHeliconia commented:
> Reviewing commit history I've notice that doesn't include stuff from v17.0 version, could you check? I've found at least two commits of my own authorship ([9494d70](https://github.com/OCA/maintenance/commit/9494d7080f40c60643b05879dd56a5d3542d782c) and [9a921a8](https://github.com/OCA/maintenance/commit/9a921a8576448277d073109fe776207a0c24dd74)) that are not included, and it shouldn't be merged in another one without respecting authorship.
> 
> In other words, this migration should start from the complete commit history of v17 addon version.

@dalonsod You're absolutely right — it appears I performed the migration from version 16.0. I’ll make the necessary corrections right away. I truly value and respect every contribution made by the community; this was simply a case of incorrect migration steps on my part. Thank you for your understanding and for taking the time to review it.

dalonsod committed:
[FIX] maintenance_equipment_hierarchy: complete name crashes for new childs

Before this fix, when we're trying to create a child equipment from
child equipments list view for a certain equipment, an error was raised.
This fixes it.

dalonsod commented:
Hello @BhaveshHeliconia will resume this?

DantePereyra commented:
@BhaveshHeliconia There was a fix introduced in version 17.0 via PR #495. Could you please cherry-pick commit [1379db7](https://github.com/OCA/maintenance/pull/495/commits/1379db7c658b60dc41affa71f0d2a80faae3ba59)? Also, kindly update the commit history as per @dalonsod’s suggestion.

LoisRForgeFlow commented:
@BhaveshHeliconia A kind reminder here, are you going to attend remaining comments?

BhaveshHeliconia commented:
@dalonsod @LoisRForgeFlow : Please review and let me know if everything is proper now or if I’m still missing something.

LoisRForgeFlow commented:
@dalonsod @DantePereyra Maybe you can confirm better if your requests have been attended.

dalonsod commented:
> There was a fix introduced in version 17.0 via PR #495. Could you please cherry-pick commit [1379db7](https://github.com/OCA/maintenance/pull/495/commits/1379db7c658b60dc41affa71f0d2a80faae3ba59)?

Fix added by that PR is already available here, but included into `18.0` migration commit, not cherry-picking it before, as proposed, could you check @BhaveshHeliconia ?

BhaveshHeliconia commented:
@dalonsod : Please review and let me know if everything is proper now or if I’m still missing something. Thank you!

dalonsod reviewed (approved):
:+1: thanks for later adjustments

etobella reviewed (approved):
/ocabot merge nobump
